src/Stormpath/Authc/UsernamePasswordRequest.php

```diff
@@ -22,12 +22,15 @@ class UsernamePasswordRequest implements AuthenticationRequest
     private $username;
     private $password;
     private $host;
+    private $accountStore;
 
-    public function __construct($username, $password, $host = null)
+    public function __construct($username, $password, array $options = array())
     {
-        $this->host = $host;
         $this->password = $password ? str_split($password) : array();
         $this->username = $username;
+
+        $this->host = isset($options['host']) ? $options['host'] : null;
+        $this->accountStore = isset($options['accountStore']) ? $options['accountStore'] : null;
```

Comment:
Why do you think an interface might be needed?

I thought of doing something like this:

```php
$this->accountStore = null;
if (isset($options['accountStore']))
{
    $accountStore = $options['accountStore'];
    if ($accountStore instanceof AccountStore)
    {
        $this->accountStore = $accountStore;
    }
}